protocol/rpcprovider/rewardserver/badger_db.go

```diff
@@ -16,7 +16,7 @@ type BadgerDB struct {
 	shardString  string
 	rewards      map[string]*entryWithTtl
 	db           *badger.DB
-	lock         sync.RWMutex
+	lock         sync.Mutex
```

Comment:
why change to Mutex? we can still Rlock for read and lock for write, it will be faster

Reply:
After a second look, you're right, I'll change that backl

Reply:
Fixed